File: src/main/java/tdd/hundreddoors/OneHundredDoors.java
package tdd.hundreddoors;

public class OneHundredDoors {
	private int pass;
	private boolean[] doorStates;
	private int doorCount;

	public OneHundredDoors(int doorCount) {
		this.doorCount = doorCount;
		if ( doorCount == 1 )
			 doorStates = new boolean[] {false};
		else
			 doorStates = new boolean[] {false,false};
	}

	public boolean[] getDoorStates() {
		return doorStates;
	}

	private boolean[] changeDoorStates() {
		if ( doorCount == 2 ) {
			if ( pass == 2 ) {
				doorStates[1] = !doorStates[1];
				return doorStates;
			}
			if ( pass == 1 ) {
				doorStates[0] = !doorStates[0];
				doorStates[1] = !doorStates[1];
				
				return doorStates;
			}
			return doorStates;
		}
		if ( pass == 1 ) {
			doorStates[0] = !doorStates[0];
			return doorStates;
		}
		return doorStates;
	}

	public void pass() {
		pass++;
		changeDoorStates();
	}

}

File: src/test/java/tdd/doors/OneHundredDoors.kt
package tdd.doors

class OneHundredDoors(val doorCount: Int) {
    private val doorStates = ArrayList<Boolean>()
    private var pass = 0

    init {
        for (i in 1..doorCount)
            doorStates.add(false)
    }

    fun status(): List<Boolean> {
        if (pass == 1) {
            for (i in 0..doorCount - 1)
                doorStates[i] = !doorStates[i]
            return doorStates
        }

        return doorStates
    }

    fun pass() {
        pass++
    }

}